Tests/SonarQube.TeamBuild.PreProcessor.Tests/RoslynExportProfileTests.cs

```diff
@@ -85,20 +94,31 @@ private static RoslynExportProfile LoadAndCheckXml(string xml)
         private static void AssertExpectedAdditionalFileExists(string fileName, RoslynExportProfile profile)
         {
             Assert.IsNotNull(profile.Configuration.AdditionalFiles);
-            IEnumerable<AdditionalFile> matches = profile.Configuration.AdditionalFiles.Where(f => string.Equals(fileName, f.FileName, System.StringComparison.OrdinalIgnoreCase));
-            Assert.AreNotEqual(0, matches.Count(), "Expected additional file was not found. File name: {0}", fileName);
-            Assert.AreEqual(1, matches.Count(), "Expecting only one matching file. File name: {0}", fileName);
-            Assert.IsNotNull(matches.First().Content, "File content should not be null. File name: {0}", fileName);
+            AdditionalFile[] matches = profile.Configuration.AdditionalFiles.Where(f => string.Equals(fileName, f.FileName, System.StringComparison.OrdinalIgnoreCase)).ToArray();
+            Assert.AreNotEqual(0, matches.Length, "Expected additional file was not found. File name: {0}", fileName);
+            Assert.AreEqual(1, matches.Length, "Expecting only one matching file. File name: {0}", fileName);
+            Assert.IsNotNull(matches[0].Content, "File content should not be null. File name: {0}", fileName);
         }
 
         private static void AssertExpectedPackageExists(string packageId, string version, RoslynExportProfile profile)
         {
             Assert.IsNotNull(profile.Deployment);
-            IEnumerable<NuGetPackageInfo> matches = profile.Deployment.NuGetPackages.Where(
+            NuGetPackageInfo[] matches = profile.Deployment.NuGetPackages.Where(
                 p => string.Equals(packageId, p.Id, System.StringComparison.Ordinal) &&
-                        string.Equals(version, p.Version, System.StringComparison.Ordinal));
-            Assert.AreNotEqual(0, matches.Count(), "Expected package was not found. Package: {0}, Version: {1}", packageId, version);
-            Assert.AreEqual(1, matches.Count(), "Expecting only one matching package. Package: {0}, Version: {1}", packageId, version);
+                        string.Equals(version, p.Version, System.StringComparison.Ordinal)).ToArray();
+            Assert.AreNotEqual(0, matches.Length, "Expected package was not found. Package: {0}, Version: {1}", packageId, version);
+            Assert.AreEqual(1, matches.Length, "Expecting only one matching package. Package: {0}, Version: {1}", packageId, version);
+        }
+
+        private static void AssertExpectedPluginExists(string pluginKey, string version, string staticResourceName, RoslynExportProfile profile)
+        {
+            Assert.IsNotNull(profile.Deployment);
+            Plugin[] matches = profile.Deployment.Plugins.Where(
+                p => string.Equals(pluginKey, p.Key, System.StringComparison.Ordinal) &&
```

Comment:
If we expect to use ordinal case sensitive comparison for the key, it might be worth adding the comparison type to the data class, like you did with the project key.